driver/panda_stream.c

```diff
@@ -124,26 +124,28 @@ static inline int step_index(int ix)
  * timeout, or because data transfer is (currently) complete.
  *    The interrupt status register records the following information:
  *
- *   31              8 7 6 5 4 3 2 1 0
+ *   31            9 8 7 6 5 4 3 2 1 0
  *  +-----------------+-+-+-+-+-+-+-+-+
- *  | sample count    | | | | | | | | |
+ *  | sample count  | | | | | | | | | |
  *  +-----------------+-+-+-+-+-+-+-+-+
- *                     | | | | | | | +-- Transfer complete, no more interrupts
- *                     | | | | | | +---- Experiment disarmed
- *                     | | | | | +------ Capture framing error
- *                     | | | | +-------- DMA error
- *                     | | | +---------- DMA address not written in time
- *                     | | +------------ Timeout
- *                     | +-------------- Block complete
- *                     +---------------- Ongoing DMA (used to validate unload)
+ *                   | | | | | | | | +-- Transfer complete, no more interrupts
+ *                   | | | | | | | +---- Experiment disarmed
+ *                   | | | | | | +------ Capture framing error
+ *                   | | | | | +-------- DMA error
+ *                   | | | | +---------- DMA address not written in time
+ *                   | | | +------------ Timeout
+ *                   | | +-------------- Block complete
+ *                   | +---------------- Ongoing DMA (used to validate unload)
+ *                   +------------------ Start event (used to capture timestamp)
  * Bit 1 records whether further interrupts are to be expected.  If this bit is
  * set then one of bits 4:1 is set to record the completion reason, unless the
  * experiment completed normally, in which case they're all set to zero.  The
  * sample count is in 4-byte transfers. */
 #define IRQ_STATUS_DONE(status)         ((bool) ((status) & 0x01))
-#define IRQ_STATUS_LENGTH(status)       ((size_t) (((status) >> 8) << 2))
+#define IRQ_STATUS_LENGTH(status)       ((size_t) (((status) >> 9) << 2))
 #define IRQ_STATUS_DMA_ACTIVE(status)   ((bool) ((status) & 0x80))
 #define IRQ_STATUS_COMPLETION(status)   ((size_t) (((status) >> 1) & 0x0F))
+#define IRQ_START_EVENT(status)         ((bool) (status  & 0x100))
```

Comment:
Forgot: we need brackets around `(status)` for macro safety!

Reply:
Also for consistency might be better to call this macro `IRQ_STATUS_START_EVENT` to match the other status register interrogation macros.